rm gcovr dep implement json html report

brianignacio5 opened the pull request:
## Description

Remove `gcovr` use in the extension in favor of implementing our own JSON and HTML report from toolchain gcov executables.

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Follow the [code coverage tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/code_coverage.md) or [code coverage](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/COVERAGE.md) documentation.
3. Observe results. After executing the `ESP-IDF: Add Editor Coverage` and `ESP-IDF: Get HTML Coverage Report for Project` should provide similar results.

## How has this been tested?

Manual testing following code coverage tutorial.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

radurentea reviewed (changes requested):
LGTM from a coding perspective.

However, I have the following suggestions for the design:
1. The black font on the red background in the HTML Report is quite hard too see:
![bg-red](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/6e27518e-0b95-4c99-bbe4-d196e1869328)

2. For the columns that don't have values we should leave them empty or put "0%" instead of "-%"
![novalues](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/9b1fe771-112d-4308-82b8-79d89fd64be8)

3. I think this is a bug, in old HTML Report the map value was  "low: <75.0 %" instead of "low: >= 0%"
![comparison](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/a7c45507-5fdd-416d-8e4e-c5b6dfd048ee)
![old comparison](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/97ef120d-1977-407a-abad-ceeb05d74355)

4. We should update the screenshots as well for the gcov documentation

radurentea reviewed (approved):
LGTM

AndriiFilippov commented:
@brianignacio5 hi ! 

Tested: 
OS - MacOS 
ESP-IDF: v5.1.2 


LGTM 👍 
Since you have updated Report, do you think[code coverage](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/COVERAGE.md) should be also updated ?

brianignacio5 commented:
> @brianignacio5 hi !
> 
> Tested: OS - MacOS ESP-IDF: v5.1.2
> 
> LGTM 👍 Since you have updated Report, do you think [code coverage tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/code_coverage.md) be also updated ? I mean like illustration.

Updated the HTML report picture and removed gcovr from docs

AndriiFilippov commented:
LGTM 👍
